### PR TITLE
Reenable the 'use auto property' lightbulb.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
A couple of changes have been made here. We no longer implement this feature as a
'compilation end' analyzer. Instead, it's a 'symbol analyzer'. This should ideally
mean that the results it produces are available much quicker since we don't have to
wait for the compiltion analysis to end.

Previously this was a compilation analyzer because we wanted to also analyze all
usages to see if switching to an auto-prop would break things. Now we eagerly
detect that you can convert to an auto prop, but we defer the conflict check
to when you try to apply the feature (like we do for "inline temp").

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
retest linux please

**Comment by CyrusNajmabadi** [comment]
retest mac please

**Comment by Pilchie** [comment]
What's the difference between this and #5722?  Should that be closed in favor of this?

**Comment by CyrusNajmabadi** [comment]
They're different approaches (one tries to avoid creating/calling into a semantic model).  We're trying to find out why we're seeing so many perf problems.  @mavasani @KevinH-MS and I are looking into things, and we're using these different PRs to investigate.

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
retest prtest/lin/dbg/unit32 please

**Comment by CyrusNajmabadi** [comment]
retest prtest/lin/dbg/unit32  please

File: src/EditorFeatures/CSharp/UseAutoProperty/CSharpUseAutoPropertyService.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                {
+                    parent = parent.Parent as ExpressionSyntax;
+                }
+
+                if (parent != null && 
+                    parent.Parent.IsKind(SyntaxKind.Argument) && 
+                    ((ArgumentSyntax)parent.Parent).RefOrOutKeyword.Kind() != SyntaxKind.None)
+                {
+                    // We accessed the field in a ref/out location.  Add a conflict annotation so the
+                    // usre knows there's a problem here.
```
Comment (bkoelman) [inline src/EditorFeatures/CSharp/UseAutoProperty/CSharpUseAutoPropertyService.cs]: user (spelling error)

**Pull request closed without merging**